amazon-chime-sdk/src/main/java/com/amazonaws/services/chime/sdk/meetings/ingestion/DefaultMeetingEventBuffer.kt

```
@@ -139,13 +139,14 @@ class DefaultMeetingEventBuffer @JvmOverloads constructor(
             // 2. If failed to send, check ttl and delete
             // 3. If succeeded, remove dirtyEvents
 
-            var dirtyEvents =
-                dirtyEventDao.listDirtyMeetingEventItems(ingestionConfiguration.flushSize)
-            var ingestionRecord =
-                IngestionEventConverter.fromDirtyMeetingEventItems(dirtyEvents, ingestionConfiguration)
-
-            var isSentSuccessful = true
             try {
+                var dirtyEvents =
+                    dirtyEventDao.listDirtyMeetingEventItems(ingestionConfiguration.flushSize)
+                var ingestionRecord =
+                    IngestionEventConverter.fromDirtyMeetingEventItems(dirtyEvents, ingestionConfiguration)
+
+                var isSentSuccessful = true
```

Will these statements throw exception?

---

The crash is from those lines, `listDirtyMeetingEventItems` specifically. Moving them into `try` because we are already wrapping the same code between line 160-165.

However, we are not swallowing the exception here, the exceptions are being swallowed in the custom Gson deserializer in `EventTypeConverters`, so that data will not be stuck in the local database forever, and they will be deleted at line 153 once they are expired.